src/VegaWrapper2.js

```diff
@@ -0,0 +1,403 @@
+
+                if (urlObj.lang) {
+                    urlParts.query.lang = urlObj.lang;
+                }
+
+                break;
+
+            default:
+                throw new Error('Unknown protocol ' + urlObj.type);
+        }
+
+        return this.formatUrl(urlParts, options);
+    };
+
+    /**
+     * Parses the response from MW Api, throwing an error or logging warnings
+     */
+    parseMWApiResponse(data) {
+        data = JSON.parse(data);
+        if (data.error) {
+            throw new Error('API error: ' + JSON.stringify(data.error));
+        }
+        if (data.warnings) {
+            this.logger('API warnings: ' + JSON.stringify(data.warnings));
+        }
+        return data;
+    };
+
+    /**
+     * For tabular and map data, extract necessary metadata from the original data
+     */
+    getMetaData(data) {
+        return [{
+            description: data.description,
+            license_code: data.license.code,
+            license_text: data.license.text,
+            license_url: data.license.url,
+            sources: data.sources
+        }];
+    }
+
+    /**
+     * Performs post-processing of the data requested by the graph's spec
+     */
+    parseResponse(data, type) {
+        switch (type) {
+            case 'wikiapi':
+                data = this.parseMWApiResponse(data);
+                break;
+            case 'wikiraw':
+                data = this.parseMWApiResponse(data);
+                try {
+                    data = data.query.pages[0].revisions[0].content;
+                } catch (e) {
+                    throw new Error('Page content not available\n' + JSON.stringify(data));
+                }
+                break;
+            case 'wikidatasparql':
+                data = JSON.parse(data);
+                if (!data.results || !Array.isArray(data.results.bindings)) {
+                    throw new Error('SPARQL query result does not have "results.bindings"');
+                }
+                data = data.results.bindings.map(row => {
+                    let key, result = {};
+                    for (key in row) {
+                        if (row.hasOwnProperty(key)) {
+                            result[key] = parseWikidataValue(row[key]);
+                        }
+                    }
+                    return result;
+                });
+                break;
+            case 'tabular':
+                data = this.parseMWApiResponse(data).jsondata;
+                const fields = data.schema.fields.map(v => {
+                    return v.name;
+                });
+                data = {
+                    meta: this.getMetaData(data),
+                    fields: data.schema.fields,
+                    data: data.data.map(v => {
+                        let row = {}, i;
+                        for (i = 0; i < fields.length; i++) {
```

Comment:
new es6 declaration should be closer to usage (and delete it above)
```suggestion
                        for (let i = 0; i < fields.length; i++) {
```